src/clims/services/workbatch.py

```diff
@@ -12,3 +15,73 @@ def register_step_template(self, step_template_cls):
     def get_step_template(self, name):
         from clims.utils import single
         return single([step for step in self.step_templates if step.name == name])
+
+
+class WorkBatch(object):
+    def __init__(self, work_batch_model):
+        self.archetype = work_batch_model
```

Comment:
Looks cool all of this.

I think we might consider making this into an Extensible, as I think you've brought up. That would make sense since it allows us to define step level properties.

I'm imagining a class structure like this:
* WorkBatch(object) <-- with wrapped models
** FragmentAnalyzeWork(WorkBatch)

But I think that can follow later!

Reply:
Yes, I have also been into that idea.